src/indexer/merger.rs

```diff
@@ -355,6 +377,55 @@ impl IndexMerger {
         }
     }
 
+    /// Checks if the readers are disjunct for their sort property and in the correct order to be
+    /// able to just stack them.
+    pub(crate) fn is_disjunct_and_sorted_on_sort_property(
+        &self,
+        sort_by_field: &IndexSortByField,
+    ) -> crate::Result<bool> {
+        let reader_and_field_accessors =
+            Self::get_reader_with_sort_field_accessor(&self.readers, sort_by_field)?;
+
+        let everything_is_in_order = reader_and_field_accessors
+            .into_iter()
+            .map(|reader| reader.1)
+            .tuple_windows()
+            .all(|(field_accessor1, field_accessor2)| {
```

Comment:
Nitpick:

No need to change anything.

I like to use `_left`, and `_right` instead of `_0`, `_1`.
The most popular alternative might be _lhs and _rhs maybe?

Reply:
in this case we want to check the order in the array, and imo `1`, `2` better expresses the elements order than left right or is this just me?

Reply:
I see what you mean. I agree!